src/containers/SubjectViewer.jsx

```diff
@@ -0,0 +1,252 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import SVGImage from '../components/SVGImage';
+import { Utility } from '../lib/Utility';
+
+import {
+  setRotation, setScaling, setTranslation,
+  setViewerState, SUBJECTVIEWER_STATE,
+} from '../ducks/subject-viewer';
+
+const INPUT_STATE = {
+  IDLE: 0,
+  ACTIVE: 1,
+}
+
+class SubjectViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    //HTML element refs.
+    this.section = undefined;
+    this.svg = undefined;
```

Comment:
I think a better default for DOM nodes or references to components is `null` instead of `undefined`.